R/vpc-simulation.R

```diff
@@ -0,0 +1,158 @@
+
+#' Create a VPC model object from an existing model
+#'
+#' @param .mod a `bbr` model object
+#' @param n number of replicates.
+#' @param seed a seed for sampling the data.
+#' @param .suffix a suffix for the VPC simulation directory. Will be prefixed by
+#'  the model id of `.mod`.
+#' @inheritParams copy_model_from
+#'
+#' @return S3 object of class `bbi_vpc_model`.
+#' @export
+new_vpc_sim <- function(
```

Comment:
As discussed offline, let's make this naming more general and call this `new_sim_model()` (and `bbi_nmsim_model` gets created by it).

We have been talking about mainly using this for VPCs, but it is not specific that use case.